pkg/config/README.md

```diff
@@ -0,0 +1,201 @@
+- **KubernetesStore**: ConfigMap-based storage for Kubernetes deployments
+- **Runtime Detection**: Automatic detection of the execution environment
+- **Singleton Pattern**: Thread-safe singleton for global configuration access
+
+## Architecture
+
+### Store Interface
+
+```go
+type Store interface {
+    Load(ctx context.Context) (*Config, error)
+    Save(ctx context.Context, config *Config) error
+    Exists(ctx context.Context) (bool, error)
+}
+```
+
+### Runtime Detection
+
+The system uses the existing `state.RuntimeDetector` to automatically choose the appropriate storage mechanism:
+
+- **Local Runtime**: Uses XDG directories for configuration files (`~/.config/toolhive/config.yaml`)
+- **Kubernetes Runtime**: Uses ConfigMaps in the pod's namespace
+
+### Configuration Structure
+
+The `Config` struct contains all application settings:
+
+```go
+type Config struct {
+    Secrets                Secrets             `yaml:"secrets"`
+    Clients                Clients             `yaml:"clients"`
+    RegistryUrl            string              `yaml:"registry_url"`
+    LocalRegistryPath      string              `yaml:"local_registry_path"`
+    AllowPrivateRegistryIp bool                `yaml:"allow_private_registry_ip"`
+    CACertificatePath      string              `yaml:"ca_certificate_path,omitempty"`
+    OTEL                   OpenTelemetryConfig `yaml:"otel,omitempty"`
+    DefaultGroupMigration  bool                `yaml:"default_group_migration,omitempty"`
+}
+```
+
+## Usage
+
+### Basic Usage
+
+```go
+// Get the global configuration (singleton)
+config := GetConfig()
+
+// Access configuration values
+registryURL := config.RegistryUrl
+secretsProvider := config.Secrets.ProviderType
+```
+
+### Updating Configuration
+
+```go
+// Update configuration with automatic storage selection
+err := UpdateConfig(func(config *Config) {
+    config.RegistryUrl = "https://new-registry.example.com"
+    config.AllowPrivateRegistryIp = true
+})
+if err != nil {
+    log.Fatalf("Failed to update config: %v", err)
+}
+```
+
+### Custom Store Usage
+
+```go
+// Create a specific store type
+store, err := NewConfigStoreWithDetector("", detector)
+if err != nil {
+    return err
+}
+
+// Load configuration
+ctx := context.Background()
+config, err := store.Load(ctx)
+if err != nil {
+    return err
+}
+
+// Modify and save
+config.RegistryUrl = "https://example.com"
+err = store.Save(ctx, config)
+```
+
+## Kubernetes Implementation
+
+### ConfigMap Structure
+
+When running in Kubernetes, the configuration is stored in server-specific ConfigMaps with the following structure:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: toolhive-config-<server-name>  # Server-specific naming (e.g., toolhive-config-fetch)
+  namespace: toolhive-system
+  labels:
+    app.kubernetes.io/name: toolhive
+    app.kubernetes.io/component: config
+data:
+  config.yaml: |
+    # YAML configuration content for this specific MCP server
```

Comment:
Not sure why we have multi-line YAML here, the `data` objects allows for YAML, is it not easier to create the ConfigMap like so:

```yaml
data:
    # YAML configuration content for this specific MCP server
    secrets:
      provider_type: "encrypted"
      setup_completed: true
    clients:
      registered_clients: []
      auto_discovery: false
    registry_url: "https://registry.example.com"
    local_registry_path: ""
    allow_private_registry_ip: false
    default_group_migration: false

```